client/src/components/AdminSignup/index.js

```diff
@@ -12,6 +14,10 @@ class AdminSignup extends Component {
     }
 
     onSubmit = e => {
+        const {
+            dispatch
+        } = this.props;
```

Comment:
In general, you pass props:
`<Component firstProp="val" />`
Then you know you'll get a prop called `firstProp`. You'll notice we also wrap some components with `connect` (e.g. `export default connect()(AdminSignup)`). This is specific to redux, and tells redux that that component needs to either dispatch actions or listen to the global state. As such, using `connect` passes `dispatch` as a prop to dispatch actions. You'll also see some components connected as follows:
```
export default connect(state => {
    return {
        user: state.user
    }
})(Nav);
```
This tells redux to inject a prop called `user` that contains the information stored in the user part of the global store (that is modified by the user reducer). When the user state changes, the prop changes, and so the component will update itself and rerender.

However, we should use prop validation which I will put in now:
https://facebook.github.io/react/docs/typechecking-with-proptypes.html